lua/entities/kleiner/init.lua

```diff
@@ -0,0 +1,970 @@
+	local ofs = (seg2.pos-self:GetPos())
+	local heightdist = ofs.z
+	local lendist = (ofs*Vector(1,1,0)):Length()
+	local jumpdir = (ofs*Vector(1,1,0)):GetNormalized()
+	local inrange = lendist < 64
+	local shouldjump = inrange and (seg3 and (seg3.area:HasAttributes(NAV_MESH_JUMP) or seg3.type == 2 or seg3.type == 3))
+	local across = (math.abs(heightdist) < 64)	
+	if(self:IsOnGround() and shouldjump)then 
+		if(across)then
+			self.loco:JumpAcrossGap((seg3 and seg3.pos) or (seg2 and seg2.pos),dir)
+			
+		else
+			self.loco:Jump() 
+		end
+	end
+	
+	--NOTE: door use seems to be pretty expensive due to the tracing, i would like to optimize this so that it doesn't run unless the path contains a doorway.
+	--Disabling the following function will just make them get stuck and teleport past the door instead
+	--[[
+	if(seg1 and seg2)then -- door handling
+		if(self.DoorPassage != nil and self.DoorPassage > CurTime())then --back away from doors if we just tried to open them
+			local dir = (seg2.pos  - seg1.pos):GetNormalized()
+	
+			self.loco:FaceTowards( IsValid(self.CurrentDoor) and self.CurrentDoor:GetPos()  )
+			self.loco:Approach(seg1.pos + dir*-32,1)
+			return false
+		end
+		local dir = (seg2.pos  - seg1.pos):GetNormalized()
+		local tr = {}
+		tr.start = self:GetPos() + Vector(0,0,32)
+		tr.endpos = tr.start + dir*64
+		tr.filter = KLEINER_NPCS_FILTER
+		tr.mask = MASK_PLAYERSOLID
+		local trace = util.TraceLine(tr)
+		debugoverlay.Line( tr.start, trace.HitPos, 0.2, Color( 255, 0, 255,0 ) ,true)
+		local blocking = trace.Entity
+		local isdoor = IsValid(blocking) and string.sub(blocking:GetClass(),6,9) == "door"
+		
+		if(IsValid(blocking) and isdoor and (!self:IsOpeningDoor() or !self:DoorIsOpen( blocking )))then
+			if(blocking.kleinerdoortime == nil or blocking.kleinerdoortime <= CurTime())then 
+				if(!self:DoorIsOpen( blocking ) )then 
+					if(blocking:HasSpawnFlags( 256 ) or string.sub(blocking:GetClass(),1,4) == "prop")then
+						blocking:Use(self,nil,SIMPLE_USE)
+					else
+						self:Teleport(seg2.pos)
+					end
+				end
+				blocking.kleinerdoortime = CurTime() + 1
+			end
+			
+			self.DoorPassage = CurTime() + 1 --stop moving path and clear doorway for a sec
+			self.CurrentDoor = blocking
+		end
+	end
+	]]
+return true
+end 
+
+
+ENT.PathGen = function( area, fromArea, ladder, elevator, length )
+	KLPATHGEN_ITERS = (KLPATHGEN_ITERS or 0) + 1
+	
+	local self = RELEVANT_KLEINER --this is bullshit, i guess this callback doesn't include the entity pathing. 
+	if(!IsValid(self))then return -1 end
+	if ( !IsValid( fromArea ) ) then
+
+		// first area in path, no cost
+		return 0
+	
+	else
+		if(KLPATHGEN_ITERS > KLPATHGEN_ITERS_BUDGET)then
+			return -1
+		end
+	
+		if ( !self.loco:IsAreaTraversable( area ) and !ladder:IsValid()) then
+			// our locomotor says we can't move here
+			return -1
+		end
+		
+		--if(!self:PosInRange(area:GetCenter()))then return -1 end
+		// compute distance traveled along path so far
```

Comment:
please do not use // comments in glua, it messes up the formatting in just about every editor

Reply:
whoops that's leftover from the original function example on the wiki